adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/credentials/BasicAuthProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.adapter.rest.credentials;
+
+import io.vertx.core.*;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.auth.User;
+import org.eclipse.hono.adapter.rest.VertxBasedRestProtocolAdapter;
+
+/**
+ * Helper class implementing {@link AuthProvider}. The Vertx components handle the header extraction and return the
+ * appropriate error codes if the user authorization fails.
+ */
+public class BasicAuthProvider implements AuthProvider {
```

Comment:
is this class supposed to be extended?

Reply:
Quote from my previous comment:



While it is generally a good idea to isolate functionality, in this case I cannot see the advantages:

- this class needs the rest adapter class passed in
- it forces that the central credentials verification method is made public for ALL adapters
- it needs some boilerplate code in the rest adapter class itself

Why not implement in this case the AuthProvider interface directly in the VertxBasedRestProtocolAdapter class?
The lines of code necessary would be reduced, and there is no need anymore to make the method public instead of protected.
WDYT?